moped-editor/src/views/projects/projectView/ProjectComponents.js

```diff
@@ -185,124 +186,107 @@ const ProjectComponents = () => {
-                    setIsEditing={false}
-                  />
-                </ErrorBoundary>
-              </Grid>
+              <ClickAwayListener onClickAway={handleComponentClickAway}>
+                <List className={classes.root}>
+                  {data.moped_proj_components.map((component, compIndex) => {
+                    const projComponentId = component.project_component_id;
+                    return (
+                      <ListItem
+                        key={"mcListItem-" + projComponentId}
+                        role={undefined}
+                        dense
+                        button
+                        onClick={() => handleComponentClick(projComponentId)}
+                        className={
+                          projComponentId === selectedComp
+                            ? classes.componentItemBlue
+                            : classes.componentItem
+                        }
+                      >
+                        <ListItemIcon>
+                          <Checkbox
+                            edge="start"
+                            className={classes.listItemCheckbox}
+                            checked={projComponentId === selectedComp}
+                            tabIndex={-1}
+                            disableRipple
+                            inputProps={{ "aria-labelledby": null }}
+                          />
+                        </ListItemIcon>
+                        <ListItemText
+                          id={"mcListItemText-" + projComponentId}
+                          primary={component?.moped_components?.component_type}
+                          secondary={
+                            (component?.moped_components?.component_subtype ??
+                              "") +
+                            [
+                              ...new Set(
+                                component.moped_proj_components_subcomponents.map(
+                                  mpcs =>
+                                    mpcs.moped_subcomponent.subcomponent_name
+                                )
+                              ),
+                            ]
+                              .sort()
+                              .join(", ")
+                          }
+                        />
+                        <ListItemSecondaryAction
+                          onClick={() => {
+                            handleComponentClick(projComponentId);
+                            handleComponentDetailsClick();
+                          }}
+                        >
+                          <IconButton edge="end" aria-label="comments">
+                            <ArrowForwardIos />
+                          </IconButton>
+                        </ListItemSecondaryAction>
+                      </ListItem>
+                    );
+                  })}
+                </List>
+              </ClickAwayListener>
```

Comment:
This is where we pass the List components (including the ClickAwayListener) directly as children of the map component.